sdk/identity/Azure.Identity/src/IManagedIdentitySource.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Identity
+{
+    internal interface IManagedIdentitySource
```

Comment:
This is as good as I was able to make it. There are still code pieces that are mostly common, like extracting json properties, so I've left them in the `ManagedIdentityClient`